game-core/src/main/java/games/strategy/triplea/attachments/CanalAttachment.java

```diff
@@ -31,24 +32,29 @@ public CanalAttachment(final String name, final Attachable attachable, final Gam
     super(name, attachable, gameData);
   }
 
-  public static Set<Territory> getAllCanalSeaZones(final String canalName, final GameData data) {
```

Comment:
This method was one of the big performance problems as it checked every territory for canal attachments. So it was replaced with a method to pass in the route to check for the given canal.